stl/inc/spanstream

```diff
@@ -217,8 +217,8 @@ public:
         requires (
             !convertible_to<_ReadOnlyRange, _STD span<_Elem>> && convertible_to<_ReadOnlyRange, _STD span<const _Elem>>)
     explicit basic_ispanstream(_ReadOnlyRange&& _Range)
-        : basic_ispanstream(
-            _STD span<_Elem>{const_cast<_Elem*>(_RANGES data(_Range)), static_cast<size_t>(_RANGES size(_Range))}) {}
+        : basic_ispanstream(_STD span<_Elem>{const_cast<_Elem*>(static_cast<_STD span<const _Elem>>(_Range).data()),
+            static_cast<_STD span<const _Elem>>(_Range).size()}) {}

```

Comment:
This converts the argument twice, and without using `forward` to restore the value category, both of which are user-observable and not the intent of https://eel.is/c++draft/ispanstream.cons#4. I think we need another `explicit` non-template constructor taking `span<const _Elem>` for this constructor template to delegate to. Something like:
```suggestion
        : basic_ispanstream(static_cast<_STD span<const _Elem>>(_STD forward<_ReadOnlyRange>(_Range))) {}

    explicit basic_ispanstream(_STD span<const _Elem> _Span) : basic_ispanstream(_STD span<_Elem>{const_cast<_Elem*>(_Span.data()), _Span.size()}) {}
```
(Evidence that adding this constructor won't introduce ambiguity: https://www.godbolt.org/z/6KT7fqKof.)